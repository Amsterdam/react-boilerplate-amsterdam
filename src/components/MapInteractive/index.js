import React from 'react';
import PropTypes from 'prop-types';
import { isEqual } from 'lodash';

import amaps from '../../static/pointquery.iife';

import './style.scss';

const DEFAULT_ZOOM_LEVEL = 14;
const PREVIEW_ZOOM_LEVEL = 16;

class Map extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      latlng: props.latlng,
    };
  }

  componentDidMount() {
    const options = {
      layer: 'standaard',
      target: 'mapdiv',
      marker: false,
      search: true,
      zoom: DEFAULT_ZOOM_LEVEL,
      onQueryResult: this.props.onQueryResult
    };

    this.map = amaps.createMap(options);
  }

  componentWillReceiveProps(props) {
    if (!isEqual(props.latlng, this.props.latlng)) {
      const latlng = new window.L.LatLng(props.latlng.latitude, props.latlng.longitude);
      this.map.then((map) => {
        map.setView(latlng, PREVIEW_ZOOM_LEVEL);
      });
    }
  }

  render() {
    return (
      <div className="map-component">
        <div className="row">
          <div className="col-12">
            <div className="map">
              <div id="mapdiv" />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Map.defaultProps = {
  latlng: {},
  onQueryResult: () => {}
};

Map.propTypes = {
  latlng: PropTypes.object,
  onQueryResult: PropTypes.func
};

export default Map;
