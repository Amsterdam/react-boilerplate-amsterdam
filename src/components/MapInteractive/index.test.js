import React from 'react';
import { shallow } from 'enzyme';

import amaps from '../../static/pointquery.iife';

import MapInteractive from './index';

jest.mock('../../static/pointquery.iife');

describe('<MapInteractive />', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <MapInteractive />
    );

    wrapper.setProps({
      location: {}
    });

    expect(wrapper).toMatchSnapshot();

    expect(amaps.createMap).toHaveBeenCalledWith({
      layer: 'standaard',
      target: 'mapdiv',
      marker: false,
      search: true,
      zoom: 14,
      onQueryResult: expect.any(Function)
    });
  });

  it('should not render map when there is one already', () => {
    const wrapper = shallow(
      <MapInteractive />
    );

    wrapper.setState({
      map: {}
    });
    wrapper.setProps({
      location: {}
    });

    expect(wrapper).toMatchSnapshot();

    expect(amaps.createMap).not.toHaveBeenCalled();
  });

  it('should render an existing location correctly', () => {
    const wrapper = shallow(
      <MapInteractive />
    );

    wrapper.setProps({
      location: {
        geometrie: {
          coordinates: [4, 52]
        },
        address: {
          openbare_ruimte: 'Dam',
          huisnummer_toevoeging: 'a',
          postcode: '1000AA',
          woonplaats: 'Amsterdam'
        }
      }
    });

    expect(wrapper).toMatchSnapshot();

    expect(amaps.createMap).toHaveBeenCalledWith({
      center: {
        latitude: 52,
        longitude: 4
      },
      layer: 'standaard',
      target: 'mapdiv',
      marker: true,
      search: true,
      zoom: 14,
      onQueryResult: expect.any(Function)
    });
  });
});
