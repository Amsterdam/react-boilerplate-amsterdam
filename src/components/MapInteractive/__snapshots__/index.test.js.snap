// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapInteractive /> should not render map when there is one already 1`] = `
<div
  className="map-component"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="map"
      >
        <div
          id="mapdiv"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<MapInteractive /> should render an existing location correctly 1`] = `
<div
  className="map-component"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="map"
      >
        <div
          id="mapdiv"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<MapInteractive /> should render correctly 1`] = `
<div
  className="map-component"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="map"
      >
        <div
          id="mapdiv"
        />
      </div>
    </div>
  </div>
</div>
`;
